source/agora/common/Serializer.d

```diff
@@ -523,6 +531,18 @@ public T deserializeFull (T) (scope DeserializeDg dg,
     else static if (is(T : BitBlob!N, size_t N))
         return T(dg(T.Width));
 
+    // Static array needs to be handled before arrays
+    // Note: This might be optimizable for small types (char, ubyte)
+    else static if (is(T : E[N], E, size_t N))
+    {
+        E deserializeEntry (size_t idx) ()
+        {
+            return deserializeFull!E(dg, opts);
+        }
+        // Note: This does not allocate because `staticMap` yields a tuple
+        return [ staticMap!(deserializeEntry, StaticIota!N) ];
```

Comment:
Note to self: D truly is marvelous

Reply:
```suggestion
        return [ Repeat!(N, deserializeEntry) ];
```

Sorry!

Reply:
And change `deserializeEntry` to remove the template parameter.

Reply:
Oh super nice !